src/App.tsx

```diff
@@ -15,47 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortOrder, setSortOrder] = useState(SortType.NONE);
+
+  const reset = () => {
+    setSortOrder(SortType.NONE);
+    setReverse(false);
+  };
+
+  const getReorderedGoods = (
+    goods: string[],
+    sortType: SortType,
+    reversed: boolean,
+  ) => {
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (reversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
```

Comment:
Why is this function inside the app component?
It will be created every re-render.
Move it outside `App` component